geo/src/algorithm/linestring_segment.rs

```diff
@@ -0,0 +1,186 @@
+use crate::line_interpolate_point::LineInterpolatePoint;
+use crate::{Coord, EuclideanLength, LineString, LinesIter, MultiLineString};
+
+/// Segments a LineString into `n` LineStrings as a MultiLineString.
+/// `None` will be returned when `n` is equal to 0, `n` is larger than the
+///  number of `Line`s that make up the `LineString`, or when a point
```

Comment:
I want to make sure I understand the intent of the code before I try to evaluate it.

Am I understanding correctly that each individual segment will have the same length as all the other segments?

> `n` is larger than the number of `Line`s that make up the `LineString`

This "n is larger than the number of Lines" condition kind of surprises me. I'd expect that we'd split edges as necessary to maintain the "equal length" invariant. Is there a reason we wouldn't?

Reply:
That is correct based on how it is written right now. 
Say `x.line_iter()` has 5 lines in it, we cannot partition it beyond 5 in that case.

It could be altered to check if n is greater than the number of lines and then using `x.densify(x.euclidean_length() / n))`. I suspect that could be done by taking total line length, dividing by the desired number of segments to calculate the appropriate value for densify.